queue/queue_test.go

```diff
@@ -51,8 +50,8 @@ func TestMemoryQueueSimpleBatch(t *testing.T) {
 
 func TestQueueTypes(t *testing.T) {
 	tests := map[string]struct {
-		memSettings  *memqueue.Settings
-		diskSettings *diskqueue.Settings
+		memSettings *memqueue.Settings
+		diskConfig  *DiskConfig
```

Comment:
There are a couple of new error cases, does it make sense to unit test them as well as the cases that should lead to a valid queue being created?

Reply:
test cases added